PolyEngine/RenderingDevice/OpenGL/Src/Proxy/GLTextureDeviceProxy.cpp

```diff
@@ -114,50 +109,195 @@ void Poly::GLTextureDeviceProxy::Resize(const ScreenSize& size)
-	}
-	else if (Usage == eTextureUsageType::FONT)
-	{
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
-	}
-	else
+	glBindTexture(GL_TEXTURE_2D, 0);
+	CHECK_GL_ERR();
+}
+
+void GLTextureDeviceProxy::InitTextureLinear()
+{
+	glGenTextures(1, &TextureID);
+
+	if (TextureID <= 0)
+		throw RenderingDeviceProxyCreationFailedException();
+
+	glBindTexture(GL_TEXTURE_2D, TextureID);
+	switch (Channels)
 	{
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR_MIPMAP_LINEAR);
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAX_LEVEL, 16);
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_REPEAT);
-		glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_REPEAT);
+		case 1:
+			InternalFormat = GL_R8;
+			break;
+		case 3:
+			InternalFormat = GL_RGB8;
+			break;
+		case 4:
+			InternalFormat = GL_RGBA8;
+			break;
+		default:
+			ASSERTE(false, "Invalid Channels!");
 	}
+	Format = GetFormat(Channels);
+
+	glTexImage2D(GL_TEXTURE_2D, 0, InternalFormat, (GLsizei)Width, (GLsizei)Height, 0, Format, GL_UNSIGNED_BYTE, nullptr);
+
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_REPEAT);
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_REPEAT);
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+	glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+
+	glBindTexture(GL_TEXTURE_2D, 0);
+	CHECK_GL_ERR();
+}
+
+void GLTextureDeviceProxy::InitTextureRenderTarget()
+{
+	glGenTextures(1, &TextureID);
+
+	if (TextureID <= 0)
+		throw RenderingDeviceProxyCreationFailedException();
+
+	glBindTexture(GL_TEXTURE_2D, TextureID);
+	InternalFormat = Channels > 3 ? GL_RGBA8 : GL_RGB8;
+	Format = Channels > 3 ? GL_RGBA : GL_RGB;
+	glTexImage2D(GL_TEXTURE_2D, 0, InternalFormat, (GLsizei)Width, (GLsizei)Height, 0, Format, GL_UNSIGNED_BYTE, nullptr);
+
+	// if(InternalUsage == eInternalTextureUsageType::DEPTH_ATTACHEMENT)
+	// 	glTexImage2D(GL_TEXTURE_2D, 0, InternalFormat, (GLsizei)Width, (GLsizei)Height, 0, GL_DEPTH_COMPONENT, GL_FLOAT, nullptr);
+	// if (InternalUsage == eInternalTextureUsageType::COLOR_ATTACHEMENT)
+	// 	glTexImage2D(GL_TEXTURE_2D, 0, InternalFormat, (GLsizei)Width, (GLsizei)Height, 0, GL_RGBA, GL_FLOAT, nullptr);
+	// else
+	// 	glTexImage2D(GL_TEXTURE_2D, 0, InternalFormat, (GLsizei)Width, (GLsizei)Height, 0, InternalFormat, GL_FLOAT, nullptr);
+	// 
+	// if (InternalUsage == eInternalTextureUsageType::COLOR_ATTACHEMENT || InternalUsage == eInternalTextureUsageType::DEPTH_ATTACHEMENT)
+	// {
+	// 	glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+	// 	glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+	// }
```

Comment:
Remove commented code or provide explanation.

Reply:
Remove RenderTargets until it will be rewritten with support for current renderer?